libraries/botframework-streaming-extensions/src/WebSocket/NodeWebSocket.ts

```diff
@@ -5,98 +5,15 @@
  * Copyright (c) Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License.
  */
-import * as http from 'http';
-import * as WaterShed from 'watershed';
-import { ISocket } from '../Interfaces/ISocket';
+import WebSocket from 'ws';
 
-export class NodeWebSocket implements ISocket {
-    private readonly waterShedSocket: any;
-    private connected: boolean;
+import { IsomorphicWebSocket } from './IsomorphicWebSocket';
 
-    /// <summary>
-    /// Creates a new instance of the NodeWebSocket class.
-    /// </summary>
-    /// <param name="waterShedSocket">The WaterShed socket object to build this connection on.</param>
-    public constructor(waterShedSocket?) {
-        this.waterShedSocket = waterShedSocket;
-        this.connected = !!waterShedSocket;
-    }
-
-    /// <summary>
-    /// True if the socket is currently connected.
-    /// </summary>
-    public isConnected(): boolean {
-        return this.connected;
-    }
-
-    /// <summary>
-    /// Writes a buffer to the socket and sends it.
-    /// </summary>
-    /// <param name="buffer">The buffer of data to send across the connection.</param>
-    public write(buffer: Buffer): void {
-        this.waterShedSocket.send(buffer);
-    }
-
-    /// <summary>
-    /// Connects to the supporting socket using WebSocket protocol.
-    /// </summary>
-    /// <param name="serverAddress">The address the server is listening on.</param>
-    /// <param name="port">The port the server is listening on, defaults to 8082.</param>
-    public async connect(serverAddress, port = 8082): Promise<void> {
-    // following template from https://github.com/joyent/node-watershed#readme
-        let shed = new WaterShed.Watershed();
-        let wskey = shed.generateKey();
-        let options = {
-            port: port,
-            hostname: serverAddress,
-            headers: {
-                connection: 'upgrade',
-                'Sec-WebSocket-Key': wskey,
-                'Sec-WebSocket-Version': '13'
-            }
-        };
-        let req = http.request(options);
-        req.end();
-        req.on('upgrade', function(res, socket, head): void {
-            shed.connect(res, socket, head, wskey);
-        });
-
-        this.connected = true;
-
-        return new Promise<void>((resolve, reject): void => {
-            req.on('close', resolve);
-            req.on('error', reject);
+export class NodeWebSocket extends IsomorphicWebSocket {
+    constructor(webSocket?: WebSocket) {

```

Comment:
The adapter will need to be updated to pass in a different object when setting up a WebSocket connection.

Reply:
Do you mean `constructor` could take up object other than `WebSocket` (from `ws`)?

Reply:
Yeah, the WebSocket connection setup here https://github.com/microsoft/botbuilder-js/blob/1fe68b7472081df968d691c4b9bb69db607e419f/libraries/botbuilder-streaming-extensions/src/BotFrameworkStreamingAdapter.ts#L91 will need to be changed to pass in an object compatible with the WebSocket, instead of the WaterShed specific socket it's currently sending.

Reply:
Oh that's cute, I didn't know it did mini-previews now. It's the three line connection setup dance that begins on line 91.